src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecute.java

```diff
@@ -169,6 +173,46 @@ public void run(ActionContext context) throws Exception {
     context.getMetrics().gauge(RECORDS_PROCESSED, rows);
   }
 
+  /**
+   * Executes Query with added retry rules following:
+   * https://cloud.google.com/bigquery/sla
```

Comment:
This should also apply to BigQuery Sink  plugin and may be even replication BQ plugins? Or for the time being it is only done for execute? It would be good to put this in a different class if it is applicable for other plugins.